packages/block-editor/src/components/iframe/index.js

```diff
@@ -0,0 +1,226 @@
+
+			if ( ! cssRules ) {
+				return;
+			}
+
+			const isMatch = Array.from( cssRules ).find(
+				( { selectorText } ) =>
+					selectorText &&
+					( selectorText.includes( `.${ BODY_CLASS_NAME }` ) ||
+						selectorText.includes( `.${ BLOCK_PREFIX }` ) )
+			);
+
+			if ( isMatch && ! doc.getElementById( ownerNode.id ) ) {
+				doc.head.appendChild( ownerNode.cloneNode( true ) );
+			}
+		} );
+	}, [] );
+}
+
+/**
+ * Bubbles some event types (keydown, keypress, and dragover) to parent document
+ * document to ensure that the keyboard shortcuts and drag and drop work.
+ *
+ * Ideally, we should remove event bubbling in the future. Keyboard shortcuts
+ * should be context dependent, e.g. actions on blocks like Cmd+A should not
+ * work globally outside the block editor.
+ *
+ * @param {Document} doc Document to attach listeners to.
+ */
+function useBubbleEvents( doc ) {
+	useEffect( () => {
+		const { defaultView } = doc;
+		const { frameElement } = defaultView;
+
+		function bubbleEvent( event ) {
+			const prototype = Object.getPrototypeOf( event );
+			const constructorName = prototype.constructor.name;
+			const Constructor = window[ constructorName ];
+
+			const init = {};
+
+			for ( const key in event ) {
+				init[ key ] = event[ key ];
+			}
+
+			if ( event instanceof defaultView.MouseEvent ) {
+				const rect = frameElement.getBoundingClientRect();
+				init.clientX += rect.left;
+				init.clientY += rect.top;
+			}
+
+			const newEvent = new Constructor( event.type, init );
+			const cancelled = ! frameElement.dispatchEvent( newEvent );
+
+			if ( cancelled ) {
+				event.preventDefault();
+			}
+		}
+
+		const eventTypes = [ 'keydown', 'keypress', 'dragover' ];
+
+		for ( const name of eventTypes ) {
+			doc.addEventListener( name, bubbleEvent );
+		}
+
+		return () => {
+			for ( const name of eventTypes ) {
+				doc.removeEventListener( name, bubbleEvent );
+			}
+		};
+	}, [] );
+}
+
+/**
+ * Sets the document direction.
+ *
+ * Sets the `editor-styles-wrapper` class name on the body.
+ *
+ * Copies the `admin-color-*` class name to the body so that the admin color
+ * scheme applies to components in the iframe.
+ *
+ * @param {Document} doc Document to add class name to.
+ */
+function useBodyClassName( doc ) {
+	useEffect( () => {
+		doc.dir = document.dir;
+		doc.body.className = BODY_CLASS_NAME;
+
+		for ( const name of document.body.classList ) {
+			if ( name.startsWith( 'admin-color-' ) ) {
+				doc.body.classList.add( name );
+			}
+		}
+	}, [] );
+}
+
+/**
+ * Positions the body element so that the resize listener works correctly. We're
+ * using an absolute position here because the resize listener doesn't seem to
+ * report shrinking when the position is relative, causing the iframe not to
+ * shrink when content is removed.
+ *
+ * @see https://github.com/FezVrasta/react-resize-aware#usage
+ *
+ * @param {Document} doc Document to set styles to.
+ */
+function useResizeListenerCompat( doc ) {
```

Comment:
This one I have trouble understanding :P

Reply:
I think I'm going to remove this iframe resizing. The reason I did that was for meta boxes in the post editor. We'll just need to find a better way to do that and let the iframe be scrollable instead of expanding it. @jasmussen noticed a bug with it above.

Reply:
Removed this now.